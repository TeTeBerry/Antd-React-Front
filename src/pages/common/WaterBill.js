import React from "react";
import { Descriptions } from "antd";
import axios from "axios";

class WaterBill extends React.Component {
  constructor(props) {
    super(props);
    console.dir(props);
    this.state = {
      waterBill: [],
      meterName: props.match.params.meterName,
      memberName: "",
      fee: "",
      totalMilliters: "",
      month: ""
    };
  }

  componentDidMount() {
    this.getWaterBill();
  }

  getWaterBill() {
    axios
      .get(
        `/iot/meter/getWaterBill/?meterName=${this.state.meterName}&password=${
          this.state.password
        }`
      )
      .then(res => {
        console.log(res);
        const mm = res.data.data.month;
        var months = [
          "January",
          "February",
          "March",
          "April",
          "May",
          "June",
          "July",
          "August",
          "September",
          "October",
          "November",
          "December"
        ];
        const month = months[mm - 1] || "";
        console.log(month);
        if (res.data.code === 200) {
          if (res.data.data.meterName === this.state.meterName) {
            this.setState({
              meterName: res.data.data.meterName,
              memberName: res.data.data.memberName,
              fee: res.data.data.fee,
              totalMilliters: res.data.data.totalMilliters,
              month: month
            });
          }
        }
      })

      .catch(error => {
        console.log(error);
      });
  }

  render() {
    return (
      <Descriptions bordered title="Water Bill" border size={this.state.size}>
        <Descriptions.Item label="Meter Name">
          {this.state.meterName}
        </Descriptions.Item>
        <Descriptions.Item label="Member Name">
          {this.state.memberName}
        </Descriptions.Item>
        <Descriptions.Item label="Month">{this.state.month}</Descriptions.Item>
        <Descriptions.Item label="Fee">{this.state.fee}</Descriptions.Item>
        <Descriptions.Item label="TotalMilliters">
          {this.state.totalMilliters}
        </Descriptions.Item>
      </Descriptions>
    );
  }
}
export default WaterBill;
