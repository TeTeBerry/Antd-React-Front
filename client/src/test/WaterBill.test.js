import React from "react";
import setTest from "../setupTests";
import { shallow, render, mount } from "enzyme";
import WaterBill from "../pages/common/WaterBill";

describe("Test case for setup volume", () => {
  it("should match its empty snapshot", () => {
    const wrapper = shallow(<WaterBill />);
    expect(wrapper).toMatchSnapshot();
  });
  it("getWaterbill", () => {
    const getWaterBill = jest.fn();
    const props = {
      getWaterBill
    };
    const wrapper = mount(<WaterBill {...props} />);
    expect(wrapper.props().getWaterBill).toHaveBeenCalledTimes(0);
  });
});
