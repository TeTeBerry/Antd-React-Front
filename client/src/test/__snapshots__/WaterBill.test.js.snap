// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for setup volume should match its empty snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WaterBill />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bordered": true,
      "childrenColumnName": "children",
      "className": "",
      "columns": Array [
        Object {
          "dataIndex": "meterName",
          "title": "Meter Name",
        },
        Object {
          "dataIndex": "memberName",
          "title": "Member Name",
        },
        Object {
          "dataIndex": "month",
          "title": "Month",
        },
        Object {
          "dataIndex": "totalMilliters",
          "title": "Amount volume (totalMilliters /mL)",
        },
        Object {
          "className": "column-money",
          "dataIndex": "fee",
          "title": "Amount Fee(Bath)",
        },
      ],
      "dataSource": Array [],
      "footer": [Function],
      "indentSize": 20,
      "loading": false,
      "locale": Object {},
      "rowKey": [Function],
      "showHeader": true,
      "size": "default",
      "sortDirections": Array [
        "ascend",
        "descend",
      ],
      "title": [Function],
      "useFixedHeader": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bordered": true,
        "childrenColumnName": "children",
        "className": "",
        "columns": Array [
          Object {
            "dataIndex": "meterName",
            "title": "Meter Name",
          },
          Object {
            "dataIndex": "memberName",
            "title": "Member Name",
          },
          Object {
            "dataIndex": "month",
            "title": "Month",
          },
          Object {
            "dataIndex": "totalMilliters",
            "title": "Amount volume (totalMilliters /mL)",
          },
          Object {
            "className": "column-money",
            "dataIndex": "fee",
            "title": "Amount Fee(Bath)",
          },
        ],
        "dataSource": Array [],
        "footer": [Function],
        "indentSize": 20,
        "loading": false,
        "locale": Object {},
        "rowKey": [Function],
        "showHeader": true,
        "size": "default",
        "sortDirections": Array [
          "ascend",
          "descend",
        ],
        "title": [Function],
        "useFixedHeader": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
